### DISCUSSION TITLE
Access user info from within a plugin defined in LangChain

### DISCUSSION THREAD

**Discussion by AMMJ93** [danny-avila/LibreChat#1763] [Q&A] [answered]
Hi, I created a custom plugin in LangChainJS and want to access the current users information such as user_id, since I want to use that information when making requests to an API endpoint. How would I go about that? So retrieving the info defined at `/LibreChat/api/models/User.js` when calling the plugin. Help is much appreciated!

**Comment by danny-avila** [comment, marked as answer]
Absolutely, I believe the userId is passed to every tool on initialization already. This happens in `api/app/clients/tools/util/handleTools.js` within the `loadToolWithAuth` function.

See `api/app/clients/tools/DALL-E.js` and `api/app/clients/tools/structured/DALLE3.js` for reference, which use this value. 

If you need more user info, you will have to mess with what is passed to `loadTools` and change the connecting layers of logic accordingly.

**Comment by AMMJ93** [comment]
> Absolutely, I believe the userId is passed to every tool on initialization already. This happens in `api/app/clients/tools/util/handleTools.js` within the `loadToolWithAuth` function.
> 
> See `api/app/clients/tools/DALL-E.js` and `api/app/clients/tools/structured/DALLE3.js` for reference, which use this value.
> 
> If you need more user info, you will have to mess with what is passed to `loadTools` and change the connecting layers of logic accordingly.

Awesome, easy enough. Thank you for your quick answer 👍
